src/distilabel/llm/base.py

```diff
@@ -14,18 +14,42 @@
 
 from __future__ import annotations
 
+import multiprocessing as mp
+import queue
 import warnings
 from abc import ABC, abstractmethod
 from concurrent.futures import Future, ThreadPoolExecutor
-from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, List, Type, Union
+from ctypes import c_char
+from enum import Enum, auto
+from functools import cached_property
+from threading import Thread
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Generator,
+    List,
+    Type,
+    Union,
+)
 
+from distilabel.logger import get_logger
 from distilabel.tasks.prompt import Prompt
+from distilabel.utils.types import is_list_of_futures
 
 if TYPE_CHECKING:
     from distilabel.llm.utils import LLMOutput
     from distilabel.tasks.base import Task
     from distilabel.tasks.prompt import SupportedFormats
 
+logger = get_logger()
```

Comment:
Feel free to discard, but maybe with `LOGGER` it's easier to read that's a global var?

```suggestion
LOGGER = get_logger()
```

Reply:
I prefer `logger` tbh